src/hotspot/share/opto/escape.cpp

```diff
@@ -4830,7 +4873,27 @@ void ConnectionGraph::split_unique_types(GrowableArray<Node *>  &alloc_worklist,
         }
         mem = mem->in(MemNode::Memory);
       }
+      if (mem->Opcode() == Op_NarrowMemProj) {
+        const TypePtr* at = mem->adr_type();
+        uint idx = (uint) _compile->get_alias_index(at->is_ptr());
+        if (idx == i) {
+          // projection for a non known allocation on a non known allocation slice: can't skip over the allocation
+          if (cur == nullptr) {
+            cur = mem;
+          }
+        } else {
+          // projection for a known allocation on a non known allocation slice: skip over the allocation
```

Comment:
The thing about "non known" sounds like we should not be able to skip it... But I guess we still know they are from unrelated slices somehow?